transcom/mymove
# B 20753 int

## [B-20753](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20753)

## Summary

This is a simple UI change to allow TOO users to enter a requested pickup date for NTS-Release shipments and requested delivery dates for NTS shipments. These will be used as "handle-in/out" dates for the prime to pick up items from or deposit items into long term storage, respectively. They are not customer facing, and will only be seen by office users/prime and only edited by TOOs. No locations are provided as NTS/rs will use facility addresses as the corresponding location with these dates. 20752 will follow shortly to display this information on move details.


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. First, create a customer and move with NTS and NTS-release shipments, one of each should suffice.
2. Fill out the information and submit the move.
3. Due to multi-role office users having permissions for all roles, we will need to test with SC and TOO roles specifically, not multi-role users.
4. Log in as a Service Counselor, navigate to your move. 
5. Edit the NTS shipment. Ensure that there is no delivery date field.
6. Edit the NTS-release shipment. Ensure that there is no pickup date field.
7. Now that we've validated the fields won't show for other office roles, let's edit these fields as a TOO.
8. Sign back into the office app as a TOO. 
9. Edit the NTS shipment. Ensure that the delivery date field is present. 
10. Enter a date and save. The date should persist after saving if you navigate back to the edit shipment page.
11. Edit the NTS-release shipment. Ensure that the pickup date field is present.
12. Enter a date and save. The date should persist after saving if you navigate back to the edit shipment page.

Note: Upon adding a requested pickup date for NTS-releases, an assumption in our calculation function caused NTS-releases to not be approvable when it attempted to price based on the non-existent pickup address (we use facility address instead). Please ensure that the NTS-release can be submitted to the prime successfully. 
### Frontend


## Screenshots
<img width="753" alt="image" src="https://github.com/user-attachments/assets/888fd5cd-7408-4798-976f-632dab1a70e6">
<img width="753" alt="image" src="https://github.com/user-attachments/assets/84ea9cda-5e9d-4440-a2a7-f7944b097150">
